src/services/render/index.js

```diff
@@ -98,20 +115,28 @@ export const renderElement = (
   if (element.nodeType === NODE_TYPE.TEXT_NODE) {
     // Render non-empty text nodes, when wrapped inside a <text> element
     if (element.nodeValue) {
-      const trimmedValue = element.nodeValue.trim();
-      if (trimmedValue.length > 0) {
-        if (
-          (element.parentNode?.namespaceURI === Namespaces.HYPERVIEW &&
-            element.parentNode?.localName === LOCAL_NAME.TEXT) ||
-          element.parentNode?.namespaceURI !== Namespaces.HYPERVIEW
-        ) {
-          return options.preformatted
-            ? trimmedValue
-            : trimmedValue.replace(/\s+/g, ' ');
+      if (
+        (element.parentNode?.namespaceURI === Namespaces.HYPERVIEW &&
+          element.parentNode?.localName === LOCAL_NAME.TEXT) ||
+        element.parentNode?.namespaceURI !== Namespaces.HYPERVIEW
+      ) {
+        if (options.preformatted) {
+          return element.nodeValue;
```

Comment:
So is it true that previously, we were (incorrectly) trimming a pre-formatted string?

Reply:
correct, the trimming was happening before the check for pre-formatted.